### PR TITLE
Add canary workflow

### PR DESCRIPTION
**Pull request by ChristianMurphy**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Follow up to: https://github.com/unifiedjs/ideas/issues/7

> Unified is a large and highly connected community of packages.
> Changes to any package could have impact downstream packages.
> Currently anticipating impact is a purely manual process.

> It could be helpful to use downstream packages, which themselves have comprehensive test suites, to help automatically detect what changes may be breaking or have unintended impacts.

### CODE REVIEW

**Comment by ChristianMurphy** [comment]
We may have caught something with `to-vfile`

```
lib/index.js(43,57): error TS2345: Argument of type 'VFileOptions | Uint8Array' is not assignable to parameter of type 'VFileCompatible'.
  Type 'Uint8Array' is not assignable to type 'VFileCompatible'.
    Type 'Uint8Array' is missing the following properties from type 'Buffer': write, toJSON, equals, compare, and 48 more.
```
https://github.com/vfile/vfile/runs/4669881846?check_suite_focus=true#step:6:21

suggests we may have a downstream compatibility issue, potentially related to #64 /cc @remcohaszing

**Comment by ChristianMurphy** [comment]
@wooorm is https://github.com/vfile/vfile-reporter-json still supported?
The repo is not currently archived, but:
```
/home/runner/work/vfile/vfile/canary/vfile/vfile-reporter-json/node_modules/to-vfile/lib/core.js:4
var vfile = require('vfile')
            ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/vfile/vfile/node_modules/vfile/index.js from /home/runner/work/vfile/vfile/canary/vfile/vfile-reporter-json/node_modules/to-vfile/lib/core.js not supported.
```

https://github.com/vfile/vfile/runs/4669881876?check_suite_focus=true#step:6:35

suggests it isn't compatible with the latest vfile :thinking:

**Comment by ChristianMurphy** [comment]
rehype issue is potentially resolved by https://github.com/rehypejs/rehype/pull/82
to-vfile issue is potentially resolved by https://github.com/vfile/to-vfile/pull/20

**Comment by wooorm** [comment]
We had a conversation about `vfile-reporter-json` a while back, it’s slated to be archived indeed due to low use.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,3 +22,47 @@ jobs:
...
+        package:
+          - vfile/to-vfile
+          - vfile/vfile-location
+          - vfile/vfile-reporter-json
+          - unifiedjs/unified
+          - remarkjs/remark
+          - rehypejs/rehype
+          - retextjs/retext
+          - remarkjs/react-markdown
+          - mdx-js/mdx
```
Comment (wooorm) [inline .github/workflows/main.yml:68]: Some of these packages are either not popular or don’t really test vfiles much. Some other packages do. How about:

```
to-vfile
vfile-location
vfile-reporter
vfile-find-up
vfile-find-down
unified
```

Comment (ChristianMurphy) [reply .github/workflows/main.yml]: updated in 05cc27d

**Comment by ChristianMurphy** [comment]
build failures should be resolved by https://github.com/vfile/to-vfile/pull/20, vfile/vfile-find-up#4, vfile/vfile-find-down#5, and vfile/vfile-location#8

**Comment by wooorm** [comment]
Nice!

**Pull request merged**